### PR TITLE
Fix incorrect dependency on xds_url_map_testcase for client config logic

### PR DESCRIPTION
**Pull request by sergiitk**
Some tests incorrectly import `framework.xds_url_map_testcase` to get `UNARY_CALL` , `EMPTY_CALL` constants. This should have never been the case: csds logic and helper should be `framework.rpc.grpc_testing`.

This PR:
1. Moves `xds_url_map_testcase.RpcType*Call` to `grpc_testing.RPC_TYPE_*_CALL`.
2. Adds `configure_unary`, `configure_empty`, `configure_rpc_type` helpers to XdsUpdateClientConfigureServiceClient.
3. Adds `RPC_TYPES_BOTH_CALLS` constant import to simplify method calls where both call types are expected.
4. Updates all tests to import constants directly from `grpc_testing`.

Similar to https://github.com/grpc/psm-interop/pull/47.

### CODE REVIEW

File: framework/rpc/grpc_testing.py
Comment (sergiitk) [inline framework/rpc/grpc_testing.py:1]: - [ ] [grpc/java/master/branch/psm-security](https://source.cloud.google.com/results/invocations/26b2564d-ce77-444e-9651-59bcedf3a8e6)
- [ ] [grpc/java/master/branch/xds_k8s_lb](https://source.cloud.google.com/results/invocations/5b3c6636-b9ab-4933-a094-9fb881b854c7)
- [ ] [grpc/java/master/branch/xds_url_map](https://source.cloud.google.com/results/invocations/269a5e9b-dade-4b5e-a83e-40c26df8ec68)
- [ ] [grpc/core/v1.62.x/branch/linux/psm-csm](https://source.cloud.google.com/results/invocations/b52f9e19-7a7c-44ff-a191-fd24eab928ac)

File: framework/rpc/grpc_testing.py [hunk 1]
```diff
@@ -130,14 +150,57 @@ def configure(
...
             deadline_sec=timeout_sec,
             log_level=logging.INFO,
         )
 
+    def configure_unary(
+        self,
+        *,
+        metadata: Optional[ConfigureMetadata] = None,
+        app_timeout: Optional[int] = None,
+        timeout_sec: int = CONFIGURE_TIMEOUT_SEC,
```
Comment (XuanWang-Amos) [inline framework/rpc/grpc_testing.py:166]: This should also be Optional? Same question for `configure_empty` and `configure_rpc_type`

File: framework/rpc/grpc_testing.py [hunk 2]
```diff
@@ -130,14 +150,57 @@ def configure(
...
+    ) -> None:
+        self.configure(
+            rpc_types=(RPC_TYPE_EMPTY_CALL,),
+            metadata=metadata,
+            app_timeout=app_timeout,
+            timeout_sec=timeout_sec,
+        )
+
+    def configure_rpc_type(
```
Comment (XuanWang-Amos) [inline framework/rpc/grpc_testing.py:189]: Can we just use this method and pass `rpc_type` for `RPC_TYPE_UNARY_CALL` and `RPC_TYPE_EMPTY_CALL`?

Comment (sergiitk) [reply framework/rpc/grpc_testing.py:189]: I think I'll just get rid of it. We only support Unary and Empty anyway.

File: framework/rpc/grpc_testing.py (hunk 1 above)
Comment (sergiitk) [reply framework/rpc/grpc_testing.py:166]: Yes, you're right.

Comment (sergiitk) [reply framework/rpc/grpc_testing.py]: Fixed

File: framework/rpc/grpc_testing.py (hunk 2 above)
Comment (sergiitk) [reply framework/rpc/grpc_testing.py]: Removed

**Review by XuanWang-Amos — APPROVED** [review]

**Pull request merged**